### PR TITLE
Profile Screen Enhancements and Offline Audio Recording Support

### PR DESCRIPTION
**Pull request by helkhour**
This PR is not final but rather a constructive one to integrate and build upon previous work, with several key improvements made to the Profile Screen and offline audio recording functionality.

Changes Made:

**Offline Audio Recording Support:**
-  Added isOffline argument to SpeakingRepositoryRecord and AudioRecorder to detect Wi-Fi connection status.
-  When offline, recordings are now saved to the device's filesDir, and when online, they are saved to the cacheDir. This ensures proper handling of storage depending on the user's connectivity.

 **UserProfileViewModel Updates:**
-  Added support for saving and retrieving offline audio recordings in the UserProfileViewModel, allowing users to view their recordings on the Profile Screen.

 **Play Audio Function:**
- Created the playAudio function to enable audio playback within the Profile Screen. This function will be used to play saved recordings when the user interacts with them. This method might be replaced by the one elaborated by @Rolph1 in his offline-mode-recording branch.

**Profile Screen UI Improvements:**
- Updated the Profile Screen to use Material Design colors for better visual consistency.
- Modified the layout and design of the Profile Screen to align with the provided Figma specifications. This ensures the UI matches the intended design and improves user experience.

**Offline Audio Recording Placeholder:**
- Created a placeholder for offline audio recordings, complementing the work done in the offline-mode-recording branch by @Rolph1. This placeholder ensures offline recordings are displayed correctly on the Profile Screen.

**Stats Placeholder Enhancements:**
-  Updated the stats placeholder to include the user's streak count. However, now the streak count appears next to the username and in the stats section. What do you think, should we keep it next to the username or in the StatsSection ?

image
_This is how the sceen currently looks like._ 

Also, the total time recording is still hardcoded, waiting for #184 

**Integration of Animated Visibility:**
- In order to display the graphs that @ymarc11 worked on in #184 , I thought about integrating AnimatedVisibility. When clicking on the StatsSection, the graphs page dynamically displays new content - soon to be the graphs.
        
        
Also, I'd like to focus on a discussion point, the usage of Elvis Operator in ProfileScreen:
In the current implementation of ProfileScreen, we use the Elvis operator (?:) to handle cases where userProfile is null, falling back to display a loading state. However, upon reviewing the logic, I noticed that userProfile is non-nullable in our setup and the repository automatically handles profile data loading when connected to the internet.

Given that:
    The Elvis operator on userProfile appears redundant since a profile is always expected to be displayed.
    The current implementation adds unnecessary complexity for a scenario that does not occur ! I even have a warning on this section on the code.

Would it make sense to remove the Elvis operator and simplify the logic?

I'm curious to hear your thoughts—do you think this simplification aligns with our expected behavior for the ProfileScreen, or are there edge cases I might be overlooking?

### CODE REVIEW

**Comment by helkhour** [comment]
The Failing tests in the CI are normal. Some components in the Profile Screen have been changed and tests were updated accordingly.

File: app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -232,38 +271,215 @@ fun ProfileScreen(navigationActions: NavigationActions, profileViewModel: UserPr
...
+                                  .testTag("no_recordings_text"))
+                    }
                   }
             }
+            ?: run {
+              Text(
+                  text = "Loading profile...",
+                  style = AppTypography.bodyLargeStyle,
+                  modifier = Modifier.testTag("loading_profile_text"))
+            }
```
Comment (MathisHage) [inline app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt:351]: I fact, as you already noticed, this code is never executed and can hence be removed.

**Review by MathisHage — CHANGES_REQUESTED** [review]
Overall, the code is of really great quality. The changes introduced really allow the recordings to be properly accessible.
Nevertheless, I still made some comments I would be glad to see addressed !

Regarding the streaks placement on the profile screen, my opinion is that the icon being at the right of the user's bio is a "cleaner" look. This is personal though.

Nice job !

File: app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt [hunk 2]
```diff
@@ -47,7 +51,10 @@ class AudioRecorder(
         AudioRecord(
             MediaRecorder.AudioSource.MIC, sampleRate, channelConfig, audioFormat, bufferSize)
 
-    audioFile = File(context.cacheDir, "audio_record.wav") // Recording in WAV format
+    // Save file to the appropriate directory based on offline mode
+    val saveDir = if (isOffline) context.filesDir else context.cacheDir
+    audioFile =
+        File(saveDir, "audio_record_${System.currentTimeMillis()}.wav") // Recording in WAV format
```
[the remark this answers is not in the document]
Comment (helkhour) [reply app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt:57]: After some research, I found that the phone cache can be automatically deleted in case the phone is full or in case of problems. 
Grosso modo, filesDir is the file directory of the permanent files. There are more information on this [page](https://developer.android.com/training/data-storage/app-specific) if you want.

File: app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt [hunk 3]
```diff
@@ -7,14 +7,17 @@ import android.content.Context
...
 import android.media.MediaRecorder
+import android.util.Log
 import androidx.core.app.ActivityCompat
 import java.io.ByteArrayOutputStream
 import java.io.File
 import java.io.FileOutputStream
 
 class AudioRecorder(
     private val context: Context,
+    private val isOffline: Boolean, // Pass offline state from ViewModel or activity
```
[the remark this answers is not in the document]
Comment (helkhour) [reply app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt:20]: Hello, 
The isOffline argument is a stateFlow in MainActivity. In order to preserve the "separation of privileges" MainActivity takes care of updating the value of isOffline and dynamically gives it as argument to the specific classes. 
Does this make sense to you ? I think this approach allows AudioRecorder class to be independent from the isOffline transition logic but still gets the value needed

File: app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -143,7 +175,18 @@ fun ProfileScreen(navigationActions: NavigationActions, profileViewModel: UserPr
...
-                              Modifier.fillMaxWidth(0.95f).height(AppDimensions.profileCardHeight),
+                              Modifier.fillMaxWidth(0.95f)
+                                  .height(AppDimensions.profileCardHeight)
+                                  .shadow(
+                                      AppDimensions.shadowDimension,
+                                      shape =
+                                          RoundedCornerShape(size = AppDimensions.statusBarPadding),
+                                      clip = false)
+                                  .background(
+                                      MaterialTheme.colorScheme.onSecondary,
```
[the remark this answers is not in the document]
Comment (helkhour) [reply app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt:186]: good catch

File: app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt [hunk 5]
```diff
@@ -164,14 +207,13 @@ fun ProfileScreen(navigationActions: NavigationActions, profileViewModel: UserPr
...
-                              ButtonDefaults.buttonColors(backgroundColor = AppColors.surfaceColor),
+                          colors = ButtonDefaults.buttonColors(backgroundColor = colors.onSurface),
                           contentPadding = PaddingValues(0.dp)) {
                             Icon(
-                                Icons.Outlined.Edit,
+                                imageVector = Icons.Outlined.Edit,
                                 contentDescription = "Edit button",
                                 modifier = Modifier.size(AppDimensions.iconSizeMedium),
-                                tint = AppColors.surfaceColor)
+                                tint = colors.errorContainer)
```
[the remark this answers is not in the document]
Comment (helkhour) [reply app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt:216]: noted

File: app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (MathisHage) [reply app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt:20]: Ok I get it now. Thanks for the clarification.

File: app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt [hunk 6]
```diff
@@ -57,30 +66,49 @@ import androidx.compose.ui.window.Dialog
...
+  var isStatsVisible by remember { mutableStateOf(false) }
+
   // Get the context
   val context = LocalContext.current
+  // Load saved recordings when the screen is displayed
+  LaunchedEffect(Unit) { profileViewModel.loadSavedRecordings(context) }
 
   // State to control whether the profile picture dialog is open
   var isDialogOpen by remember { mutableStateOf(false) }
+  val audioRecorder = remember { AudioRecorder(context, isOffline = true) }
```
[the remark this answers is not in the document]
Comment (helkhour) [reply app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt]: good catch ! I put this value for testing and forgot to remove it. 
In fact and as said before, the isOffline argument is dynamically updated in MainActivity.

File: app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (MathisHage) [reply app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt:57]: Thanks !

File: app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt [hunk 7]
```diff
@@ -232,38 +271,215 @@ fun ProfileScreen(navigationActions: NavigationActions, profileViewModel: UserPr
...
+                  horizontal = AppDimensions.paddingMedium,
+                  vertical = AppDimensions.paddingSmall)) {
+        // Play Button Icon
+        Icon(
+            imageVector = Icons.Default.PlayArrow,
+            contentDescription = "Play",
+            modifier =
+                Modifier.size(AppDimensions.iconSizeLarge)
+                    .align(Alignment.CenterVertically)
+                    .background(colors.errorContainer, shape = CircleShape))
```
[the remark this answers is not in the document]
Comment (MathisHage) [reply app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt:466]: Hey ! Unresolving this as it is still an error color.

[the remark this answers is not in the document]
Comment (MathisHage) [reply app/src/main/java/com/github/se/orator/ui/profile/ProfileScreen.kt:466]: Hey ! Unresolving this as it is still an error color.

**Comment by Rolph1** [comment]
Hi Helen,
Since this PR relies too heavily on saving the offline recordings in a file directory in the device storage rather than in the cache like we decided on with the coaches, I will have to close this PR and open a new one where we focus solely on enhancing the profile UI. 
Your work is greatly appreciated though, and in the worst case we will open a new PR with this branch.
